xarray/core/formatting.py

```diff
@@ -180,20 +199,29 @@ def format_array_flat(array, max_width):
     array that will fit within max_width characters.
     """
     # every item will take up at least two characters, but we always want to
-    # print at least one item
-    max_possibly_relevant = max(int(np.ceil(max_width / 2.0)), 1)
-    relevant_items = first_n_items(array, max_possibly_relevant)
+    # print at least first and last items
+    max_possibly_relevant = min(max(array.size, 1),
+                                max(int(np.ceil(max_width / 2.)), 2))
+    relevant_items = \
+        sum(zip_longest(
+            first_n_items(array, (max_possibly_relevant + 1) // 2),
+            reversed(last_n_items(array, max_possibly_relevant // 2))),
+            ())[:max_possibly_relevant]
     pprint_items = format_items(relevant_items)
 
     cum_len = np.cumsum([len(s) + 1 for s in pprint_items]) - 1
-    if (max_possibly_relevant < array.size or (cum_len > max_width).any()):
-        end_padding = u' ...'
-        count = max(np.argmax((cum_len + len(end_padding)) > max_width), 1)
-        pprint_items = pprint_items[:count]
+    if (max_possibly_relevant < array.size) or (cum_len > max_width).any():
+        padding = u' ... '
+        count = min(array.size,
+                    max(np.argmax(cum_len + len(padding) - 1 > max_width), 2))
     else:
-        end_padding = u''
+        count = array.size
+        padding = u'' if (count <= 1) else u' '
 
-    pprint_str = u' '.join(pprint_items) + end_padding
+    pprint_str = u' '.join(np.take(pprint_items, range(0, count, 2))) + \
+                 padding + \
+                 u' '.join(np.take(pprint_items,
+                                   range(count - (count % 2) - 1, 0, -2)))
```

Comment:
I think this would be would be easier to understand with something like the following:

```python
relevant_front_items = first_n_items(array, (max_possibly_relevant + 1) // 2)
relevant_end_items = ...
relevant_items = sum(zip_longest(...))

num_front = ...
num_end = ...

front_items = relevant_front_items[:num_front]
end_items = relevant_end_items[-num_end:]
pprint_str = u' '.join(front_items) + padding + u' '.join(end_items)
```

I see how you currently get the front and end items back from `relevant_items`, but it is not immediately obvious that it is correct.